src/test_workflow/integ_test/integ_test_suite_opensearch_dashboards.py

```diff
@@ -81,6 +97,38 @@ def __setup_cluster_and_execute_test_config(self, config: str) -> int:
             os.chdir(self.work_dir)
             return self.execute_integtest_sh(endpoint, port, security, config)
 
+    def multi_execute_integtest_sh(self, cluster_endpoints: list, security: bool, test_config: str) -> int:
+        script = ScriptFinder.find_integ_test_script(self.component.name, self.repo.working_directory)
+
+        def custom_node_endpoint_encoder(node_endpoint: NodeEndpoint) -> dict:
+            return {"endpoint": node_endpoint.endpoint, "port": node_endpoint.port, "transport": node_endpoint.transport}
+        if os.path.exists(script):
+            single_node = cluster_endpoints[0].data_nodes[0]
+            cmd = f"bash {script} -b {single_node.endpoint} -p {single_node.port} -s {str(security).lower()} -t {self.component.name} -v {self.bundle_manifest.build.version} -o default"
```

Comment:
if `self.component.name` is empty will the tests still run for all components by default?

Reply:
The list of components is fetched [here](https://github.com/Divyaasm/opensearch-build/blob/305a8abada0d470f6d0d695b37b44c3c830c0279/src/test_workflow/integ_test/integ_test_runner.py#L42) based on input argument --component if that's not passed it runs for all the components in the test manifest. self.component.name has the plugin name and has it's value from the manifest  and is not empty in any case.  

We alter the plugin names passed to self.component.name based on --component argument to the workflow but run them individually each time with -t component.name as per test manifest.

Reply:
Thanks! This makes it clear.